sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/TLSEvaluationStage.java

```diff
@@ -0,0 +1,291 @@
+package com.mesosphere.sdk.offer.evaluate;
+
+import com.mesosphere.sdk.dcos.CertificateAuthorityClient;
+import com.mesosphere.sdk.dcos.DcosConstants;
+import com.mesosphere.sdk.dcos.SecretsClient;
+import com.mesosphere.sdk.dcos.auth.CachedTokenProvider;
+import com.mesosphere.sdk.dcos.auth.ServiceAccountIAMTokenProvider;
+import com.mesosphere.sdk.dcos.auth.TokenProvider;
+import com.mesosphere.sdk.dcos.ca.DefaultCAClient;
+import com.mesosphere.sdk.dcos.http.HttpClientBuilder;
+import com.mesosphere.sdk.dcos.http.URLHelper;
+import com.mesosphere.sdk.dcos.secrets.DefaultSecretsClient;
+import com.mesosphere.sdk.dcos.secrets.Secret;
+import com.mesosphere.sdk.dcos.secrets.SecretsException;
+import com.mesosphere.sdk.offer.MesosResourcePool;
+import com.mesosphere.sdk.scheduler.SchedulerFlags;
+import org.apache.http.client.fluent.Executor;
+import org.apache.http.impl.client.LaxRedirectStrategy;
+import org.apache.mesos.Protos;
+import org.bouncycastle.asn1.pkcs.PKCSObjectIdentifiers;
+import org.bouncycastle.asn1.x500.X500Name;
+import org.bouncycastle.asn1.x500.X500NameBuilder;
+import org.bouncycastle.asn1.x500.style.BCStyle;
+import org.bouncycastle.asn1.x509.*;
+import org.bouncycastle.openssl.jcajce.JcaMiscPEMGenerator;
+import org.bouncycastle.operator.ContentSigner;
+import org.bouncycastle.operator.OperatorCreationException;
+import org.bouncycastle.operator.jcajce.JcaContentSignerBuilder;
+import org.bouncycastle.pkcs.PKCS10CertificationRequest;
+import org.bouncycastle.pkcs.PKCS10CertificationRequestBuilder;
+import org.bouncycastle.pkcs.jcajce.JcaPKCS10CertificationRequestBuilder;
+import org.bouncycastle.util.io.pem.PemReader;
+import org.bouncycastle.util.io.pem.PemWriter;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.*;
+import java.security.*;
+import java.security.cert.CertificateEncodingException;
+import java.security.cert.X509Certificate;
+import java.security.interfaces.RSAPrivateKey;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.util.*;
+
+public class TLSEvaluationStage implements OfferEvaluationStage {
+
+    private static final Logger LOGGER = (Logger) LoggerFactory.getLogger(TLSEvaluationStage.class);
```

Comment:
If there's only one instance of this class, indeed no need for static logger. +1 for cast removal.